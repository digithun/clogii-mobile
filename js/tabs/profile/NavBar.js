'use strict';

import React from 'react';
import {
  Text,
  Platform,
  View,
  StyleSheet,
  TouchableOpacity,
  Image
} from 'react-native';

import {colors} from './common';

const NAV_BAR_HEIGHT = 50;

export default class NavBar extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.leftMenu} onPress={() => this.props.onLeftPress && this.props.onLeftPress()}>
          <Image style={styles.backButton} source={require('./img/icons/backButton.png')}/>
        </TouchableOpacity>
        <View style={styles.midMenu}>
          <Text style={styles.navText}>{this.props.title}</Text>
        </View>
        <TouchableOpacity style={styles.rightMenu} onPress={() => this.props.onRightPress && this.props.onRightPress()}>
          {
            this.props.renderRightMenu ? this.props.renderRightMenu() : null
          }
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 0,
    height: NAV_BAR_HEIGHT,
    backgroundColor: 'white',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderColor: colors.greyBorder
  },
  backButton: {
    height: 20,
    resizeMode: 'contain'
  },
  midMenu: {
    alignItems: 'center'
  },
  leftMenu: {
    paddingLeft: 20,
    alignItems: 'flex-start'
  },
  rightMenu: {
    paddingRight: 20,
    alignItems: 'flex-end'
  },
  navText: {
    fontSize: 22,
    fontWeight: 'bold',
    color: 'rgb(209, 87, 92)'
  }
});
