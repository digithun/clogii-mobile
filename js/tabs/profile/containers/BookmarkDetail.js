import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import BookmarkDetail from '../components/BookmarkDetail';
import * as mockData from '../mockData';

const query = gql`
  query BookmarkDetail($id: MongoID!){
    clog(filter: { _id: $id }) {
      ...BookmarkDetail
    }
  }
  ${BookmarkDetail.fragments.clog}
`;

const mapQueryToProps = ({ data }) => {
  return {
    clog: data.clog,
  };
};

export const mapPropsToOptions = ({ id }) => ({
  variables: {
    id,
  },
});

export default graphql(
  query,
  {
    props: mapQueryToProps,
    options: mapPropsToOptions,
  },
)(BookmarkDetail);
