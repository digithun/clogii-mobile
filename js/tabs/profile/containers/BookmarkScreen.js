import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import BookmarkScreen from '../components/BookmarkScreen';
import * as mockData from '../mockData';

export const query = gql`
  query {
    me {
      id
      summaryBookmarks {
        ...BookmarkScreen
      }
    }
  }
  ${BookmarkScreen.fragments.summaryBookmark}
`;

const mapQueryToProps = ({ data }) => {
  return {
    summaryBookmarks: !data.me || !data.me.summaryBookmarks ? [] : data.me.summaryBookmarks,
  };
};

export default graphql(
  query,
  {
    props: mapQueryToProps,
  }
)(BookmarkScreen);
