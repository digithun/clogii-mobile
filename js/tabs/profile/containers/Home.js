import { connect } from 'react-redux';
import Home from '../components/Home';
import * as mockData from '../mockData';

export default connect(state => ({ user: state.user, followingCount: 1800, followerCount: 1800, candys: 100 }))(Home);
