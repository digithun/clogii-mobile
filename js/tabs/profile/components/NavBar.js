

import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
} from 'react-native';

import { NavBarWithPinkButton } from '../../../common/NavBar';

import { colors } from '../../../common/styles';

export default class NavBar extends React.Component {
  render() {
    return (
      <NavBarWithPinkButton
        {...this.props}
        containerStyle={styles.container}
        renderRightMenu={this.renderRightMenu.bind(this)}
        title={this.props.title}
        titleTextStyle={styles.titleText}
      />
    );
  }

  renderRightMenu() {
    return (
      <TouchableOpacity style={styles.rightMenu} onPress={this.props.onRightPress}>
        {
          this.props.renderRightMenu ? this.props.renderRightMenu() : null
        }
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderColor: colors.greyBorder,
  },
  backButton: {
    height: 20,
    resizeMode: 'contain',
  },
  titleText: {
    fontSize: 22,
    fontWeight: 'bold',
    color: colors.textPink,
  },
});
