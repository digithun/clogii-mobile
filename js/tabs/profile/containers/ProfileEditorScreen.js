import React from 'react';

import {
  Image,
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  TextInput,
  Switch,
  Alert,
  InteractionManager
} from 'react-native';
import moment from 'moment';
import Picker from 'react-native-picker';
import ImagePicker from 'react-native-image-picker';

import {changePublicProfile} from '../../../actions/changeProfile';
import {linkFacebook, unlinkFacebook} from '../../../actions/login';

import NavBar from '../components/NavBar';
import {styles as commonStyles} from '../common';
import {colors as commonColors} from '../../../common/styles';

import ProfileHeader from '../components/ProfileHeader';
import DatePickerDialog from '../components/DatePickerDialog';
import ModalSpinner from '../components/ModalSpinner';

import {connect} from 'react-redux';

export const OKButton = () => (<Image
      source={require('../img/icons/ok.png')}
      style={{
        height: 25,
        width: 25,
        resizeMode: 'contain'
      }}
    />);

class ProfileEditorScreen extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      name: this.props.name,
      birthDayDate: this.props.birthDayDate,
      sex: this.props.sex,
      changedProfilePicture: null,
      changedProfileCover: null,
      savingProfile: false,
      linkingFacebook: false
    };
  }

  render() {
    return (<View style={commonStyles.listViewContainer}>
      <ModalSpinner visible={this.state.savingProfile || this.state.linkingFacebook}/>
      <DatePickerDialog ref="datePicker"/>
      <NavBar
        title="แก้ไขข้อมูล"
        renderRightMenu={OKButton}
        onLeftPress={this.props.onBackPress}
        onRightPress={() => {
          this.setState({
            savingProfile: true
          });
          return this.props.changePublicProfile(this.state.name, this.state.birthDayDate, this.state.sex, this.state.changedProfilePicture, this.state.changedProfileCover)
            .then(() => {
              this.setState({
                savingProfile: false
              });
              this.props.onBackPress && this.props.onBackPress();
            }).catch(error => {
              Alert.alert("Error", error.message, [
                { text: "OK", onPress: () => this.setState({ savingProfile: false}) }
              ]);
            });
        }}
        />
      <ProfileHeader
        user={this.props.user}
        customCoverSource={this.state.changedProfileCover ? {uri: 'data:image/jpeg;base64,' + this.state.changedProfileCover.data, isStatic: true} : null }
        customSource={this.state.changedProfilePicture ? {uri: 'data:image/jpeg;base64,' + this.state.changedProfilePicture.data, isStatic: true} : null }
        >
        <View style={styles.rowDirection}>
          <TouchableOpacity name="profileImageInput" style={styles.whiteBorder} onPress={() => this.openProfilePicker()}>
            <Text style={styles.whiteText}>เปลี่ยนรูปโปรไฟล์</Text>
          </TouchableOpacity>
          <TouchableOpacity name="coverImageInput" style={styles.whiteBorder} onPress={() => this.openProfileCoverPicker()}>
            <Text style={styles.whiteText}>เปลี่ยนรูปพื้นหลัง</Text>
          </TouchableOpacity>
        </View>
      </ProfileHeader>

      <View style={{flex: 2}}>
        <View style={[styles.editorBox, {marginVertical: 15}]}>
          <View style={styles.row}>
            <View style={styles.labelContainer}>
              <Text style={styles.label}>ชื่อ</Text>
            </View>
            <View style={[styles.input, styles.bottomBorderGrey]}>
              <TextInput
                onChangeText={(name) => this.setState({name})}
                style={[styles.textGrey, {flex: 1}]}
                autoCapitalize="none"
                value={this.state.name}
              />
            </View>
          </View>
          <View style={styles.row}>
            <View style={styles.labelContainer}>
              <Text style={styles.label}>เพศ</Text>
            </View>
            <TouchableOpacity name="sexInput" style={[styles.input, styles.bottomBorderGrey]} onPress={() => this.openSexPicker()}>
              <Text style={styles.textGrey}>{this.state.sex === 'M' ? 'ชาย' : (this.state.sex === 'F' ? 'หญิง' : 'ไม่ระบุ')}</Text>
            </TouchableOpacity>
          </View>
          <View style={styles.row}>
            <View style={styles.labelContainer}>
              <Text style={styles.label}>เกิด</Text>
            </View>
            <TouchableOpacity name="birthDayInput" style={styles.input} onPress={() => this.openDatePicker()}>
              <Text style={styles.textGrey}>{!this.state.birthDayDate ? 'ไม่ระบุ' : moment(this.state.birthDayDate).format('LL')}</Text>
            </TouchableOpacity>
          </View>
        </View>


        <View style={styles.editorBox2}>
          <TouchableOpacity style={styles.row2} onPress={this.onToggleFacebookLink.bind(this)}>
            <Text style={styles.label2}>เชื่อมต่อ Facebook</Text>
            <View>
              <Switch value={this.props.facebookLinked} onValueChange={this.onToggleFacebookLink.bind(this)}/>
            </View>
          </TouchableOpacity>
          <TouchableOpacity style={styles.row2} onPress={this.pushPage.bind(this, 'change-email')}>
            <Text style={styles.label2}>เปลี่ยน Email</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.row2} onPress={this.pushPage.bind(this, 'change-password')}>
            <Text style={styles.label2}>เปลี่ยน Password</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>);
  }

  pushPage(page) {
    this.props.navigator.push({page});
  }

  onToggleFacebookLink() {
    this.setState({
      linkingFacebook: true
    });
    if (this.props.facebookLinked) {
      return this.props.unlinkFacebook().then(() => {
        this.setState({
          linkingFacebook: false
        });
      }).catch(error => {
        Alert.alert("Error", error.message, [
          { text: "OK", onPress: () => this.setState({ linkingFacebook: false}) }
        ]);
      });
    }
    else {
      return this.props.linkFacebook().then(() => {
        this.setState({
          linkingFacebook: false
        });
      }).catch(error => {
        Alert.alert("Error", error.message, [
          { text: "OK", onPress: () => this.setState({ linkingFacebook: false}) }
        ]);
      });
    }
  }

  openProfilePicker() {
    return new Promise((resolve, reject) => {
      ImagePicker.showImagePicker(response => {
        if(!response.error && !response.didCancel) {
          this.setState({
            changedProfilePicture: response
          });
        }
        resolve();
      });
    });
  }

  openProfileCoverPicker() {
    return new Promise((resolve, reject) => {
      ImagePicker.showImagePicker(response => {
        if(!response.error && !response.didCancel) {
          this.setState({
            changedProfileCover: response
          });
        }
        resolve();
      });
    });
  }

  openDatePicker() {
    return this.refs.datePicker.open().then(date => {
      this.setState({
        birthDayDate: date
      });
    });
  }

  openSexPicker() {
    return new Promise((resolve, reject) => {
      Picker.init({
        pickerData: [
          'ไม่ระบุ',
          'ชาย',
          'หญิง'
        ],
        selectedValue: [this.state.sex === 'M' ? 'ชาย' : (this.state.sex === 'F' ? 'หญิง' : 'ไม่ระบุ')],
        pickerConfirmBtnText: 'OK',
        pickerCancelBtnText: 'CANCEL',
        pickerTitleText: 'เพศ',
        onPickerConfirm: (data) => {
          let sex;
          if (data[0] === 'ชาย') {
            sex = 'M';
          }
          if (data[0] === 'หญิง') {
            sex = 'F';
          }
          if (data[0] === 'ไม่ระบุ') {
            sex = null;
          }
          this.setState({
            sex
          });
          resolve();
        }
      });
      Picker.show();
    });
  }
}

const styles = StyleSheet.create({
  rowDirection: {
    flexDirection: 'row',
    backgroundColor: 'transparent',
    marginTop: 25
  },
  whiteBorder: {
    padding: 3,
    paddingHorizontal: 8,
    borderWidth: 1,
    borderColor: 'white',
    borderRadius: 5,
    marginHorizontal: 5
  },
  whiteText: {
    color: 'white',
    fontSize: 10
  },
  editorBox: {
    backgroundColor: 'white',
    borderWidth: 1,
    borderColor: commonColors.greyBorder
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  labelContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  label: {
    fontSize: 15
  },
  input: {
    flex: 5,
    justifyContent: 'center',
    height: 50
  },
  editorBox2: {
    backgroundColor: 'white',
    borderTopWidth: 1,
    borderColor: commonColors.greyBorder
  },
  row2: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 15,
    borderBottomWidth: 1,
    borderColor: commonColors.greyBorder
  },
  label2: {
    fontSize: 15,
    flex: 1
  },
  bottomBorderGrey: {
    borderBottomWidth: 1,
    borderColor: commonColors.greyBorder
  },
  textGrey: {
    color: commonColors.textGrey
  }
});

const select = state => ({
  name: state.user.name,
  email: state.user.email,
  sex: state.user.sex,
  birthDayDate: state.user.birthDayDate,
  saved: state.user.savedProfile,
  facebookLinked: state.user.facebookLinked
});

// saving: state.user.savingProfile || state.user.facebookLinkProcessing

const actionsMaping = {
  changePublicProfile,
  linkFacebook,
  unlinkFacebook
};

export default connect(select, actionsMaping)(ProfileEditorScreen);
export {
  ProfileEditorScreen as Component
};
