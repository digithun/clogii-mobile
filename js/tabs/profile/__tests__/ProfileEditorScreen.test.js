import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import F8Button from 'F8Button';

import {
  Switch
} from 'react-native';

import NavBar from '../components/NavBar';
import {Component as ProfileEditorScreenComponent} from '../containers/ProfileEditorScreen';

describe('ProfileEditorScreen', () => {
  function api(name, birthDayDate, sex, changedProfilePicture, changedProfileCover) {
    if (name === 'fail') {
      return Promise.reject(new Error('something wrong'));
    }
    return Promise.resolve();
  }

  it('render', () => {
    const tree = renderer.create(<ProfileEditorScreenComponent/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('call api changePublicProfile', async () => {
    let _resolve;
    let changePublicProfile = jest.fn(() => new Promise(resolve => _resolve = resolve));
    const wrapper = shallow(<ProfileEditorScreenComponent changePublicProfile={changePublicProfile}/>);
    expect(wrapper.state().savingProfile).toBe(false);
    const task = wrapper.find(NavBar).props().onRightPress();
    expect(wrapper.state().savingProfile).toBe(true);
    _resolve();
    await task;
    expect(wrapper.state().savingProfile).toBe(false);
    expect(changePublicProfile).toBeCalled();
  });

  it('call onBackPress if changed profile', async () => {
    let changePublicProfile = jest.fn(api);
    let onBackPress = jest.fn();
    const wrapper = shallow(<ProfileEditorScreenComponent changePublicProfile={changePublicProfile} onBackPress={onBackPress}/>);
    await wrapper.find(NavBar).props().onRightPress();
    expect(onBackPress).toBeCalled();
  });

  it('call api linkFacebook', async () => {
    let _resolve;
    let linkFacebook = jest.fn(() => new Promise(resolve => {
      _resolve = resolve;
    }));
    const wrapper = shallow(<ProfileEditorScreenComponent linkFacebook={linkFacebook} facebookLinked={false}/>);
    expect(wrapper.state().linkingFacebook).toBe(false);
    const linkingPromise = wrapper.find(Switch).props().onValueChange();
    expect(linkFacebook).toBeCalled();
    expect(wrapper.state().linkingFacebook).toBe(true);
    _resolve();
    await linkingPromise;
    expect(wrapper.state().linkingFacebook).toBe(false);
  });

  it('call api unlinkFacebook', async () => {
    let unlinkFacebook = jest.fn(() => Promise.resolve());
    let onBackPress = jest.fn();
    const wrapper = shallow(<ProfileEditorScreenComponent unlinkFacebook={unlinkFacebook} facebookLinked={true}/>);
    await wrapper.find(Switch).props().onValueChange();
    expect(unlinkFacebook).toBeCalled();
  });

  it('open DatePickerDialog');
  it('open SexPicker');
  it('open BirthDayPicker');
});
