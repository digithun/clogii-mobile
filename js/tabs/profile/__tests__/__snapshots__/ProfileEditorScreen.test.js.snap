exports[`ProfileEditorScreen render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(250, 250, 250)",
      "flex": 1,
    }
  }>
  <Modal
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.6)",
          "flex": 1,
          "justifyContent": "center",
        }
      }>
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "height": 80,
            },
          ]
        } />
    </View>
  </Modal>
  <Modal
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.3)",
          "flex": 1,
          "justifyContent": "center",
        }
      }>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.9)",
            "borderColor": "rgba(0, 0, 0, 0.3)",
            "borderRadius": 10,
            "borderWidth": 0.2,
            "width": 300,
          }
        }>
        <View
          style={undefined}>
          <RCTDatePicker
            date={1482898705878}
            maximumDate={undefined}
            minimumDate={undefined}
            minuteInterval={undefined}
            mode="date"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 216,
              }
            }
            timeZoneOffsetInMinutes={undefined} />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }>
          <View
            style={
              Object {
                "flex": 1,
              }
            }>
            <View
              accessibilityComponentType="button"
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {},
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#0C42FD",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }>
                  ok
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }>
            <View
              accessibilityComponentType="button"
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {},
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#0C42FD",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }>
                  cancel
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 1,
        "borderColor": "rgb(240, 240, 240)",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingTop": 5,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "opacity": 1,
          "paddingLeft": 20,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
          }
        } />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(209, 87, 92)",
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }>
        แก้ไขข้อมูล
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "opacity": 1,
          "paddingRight": 20,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
            "width": 25,
          }
        } />
    </View>
  </View>
  <Image
    customSource={null}
    source={1}
    style={
      Array [
        Object {
          "resizeMode": "cover",
        },
        Object {
          "alignItems": "center",
          "height": undefined,
          "justifyContent": "center",
          "width": undefined,
        },
      ]
    }
    user={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": undefined,
          "justifyContent": "center",
          "paddingVertical": 10,
          "width": undefined,
        }
      }>
      <View
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          }
        } />
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "marginTop": 25,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "white",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 3,
              "paddingHorizontal": 8,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 10,
              }
            }>
            เปลี่ยนรูปโปรไฟล์
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "white",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 3,
              "paddingHorizontal": 8,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 10,
              }
            }>
            เปลี่ยนรูปพื้นหลัง
          </Text>
        </View>
      </View>
    </View>
  </Image>
  <View
    style={
      Object {
        "flex": 2,
      }
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "rgb(240, 240, 240)",
            "borderWidth": 1,
          },
          Object {
            "marginVertical": 15,
          },
        ]
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
              }
            }>
            ชื่อ
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 5,
                "height": 50,
                "justifyContent": "center",
              },
              Object {
                "borderBottomWidth": 1,
                "borderColor": "rgb(240, 240, 240)",
              },
            ]
          }>
          <TextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            style={
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.5)",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            value={undefined} />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
              }
            }>
            เพศ
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "rgb(240, 240, 240)",
              "flex": 5,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.5)",
              }
            }>
            ไม่ระบุ
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
              }
            }>
            เกิด
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 5,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.5)",
              }
            }>
            ไม่ระบุ
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "rgb(240, 240, 240)",
          "borderTopWidth": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "rgb(240, 240, 240)",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
            }
          }>
          เชื่อมต่อ Facebook
        </Text>
        <View>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "rgb(240, 240, 240)",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
            }
          }>
          เปลี่ยน Email
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "rgb(240, 240, 240)",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
            }
          }>
          เปลี่ยน Password
        </Text>
      </View>
    </View>
  </View>
</View>
`;
